app/models/shipit/task.rb

```diff
@@ -231,6 +235,10 @@ def supports_rollback?
       false
     end
 
+    def previous_success
```

Comment:
Are this  and `previous_successful` on the class too similarly named? Could they be the same?

Reply:
They can be named the same, yeah. I wasn't sure if there would be problems in having similarly named class and instance methods, but now that I have tests, it's simple to experiment and see that it works.